bootstrap/remote/gc.py

```
@@ -45,25 +59,82 @@ def collect_reachable(self) -> ReachabilitySet:
+        except FileNotFoundError:
+            pass
+
+    def _load_head_history(self, head_hash: str):
+        from bootstrap.remote.models import ServerHistory
+
+        history_path = generation_dir(self.root, head_hash) / "history.pb2"
+        if not history_path.is_file():
+            return None
+        try:
+            return read_pb2(history_path, ServerHistory)
+        except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)